ddl/failtest/fail_db_serial_test.go

```diff
@@ -340,103 +317,125 @@ func (s *testFailDBSuite) TestGenGlobalIDFail(c *C) {
 	tk.MustExec("admin check table t2")
 }
 
-func batchInsert(tk *testkit.TestKit, tbl string, start, end int) {
-	dml := fmt.Sprintf("insert into %s values", tbl)
-	for i := start; i < end; i++ {
-		dml += fmt.Sprintf("(%d, %d, %d)", i, i, i)
-		if i != end-1 {
-			dml += ","
-		}
-	}
-	tk.MustExec(dml)
-}
-
-func (s *testFailDBSuite) TestAddIndexWorkerNum(c *C) {
-	tk := testkit.NewTestKit(c, s.store)
-	tk.MustExec("create database if not exists test_db")
-	tk.MustExec("use test_db")
-	tk.MustExec("drop table if exists test_add_index")
-	if s.IsCommonHandle {
-		tk.Se.GetSessionVars().EnableClusteredIndex = variable.ClusteredIndexDefModeOn
-		tk.MustExec("create table test_add_index (c1 bigint, c2 bigint, c3 bigint, primary key(c1, c3))")
-	} else {
-		tk.MustExec("create table test_add_index (c1 bigint, c2 bigint, c3 bigint, primary key(c1))")
-	}
-
-	done := make(chan error, 1)
-	start := -10
-	// first add some rows
-	for i := start; i < 4090; i += 100 {
-		batchInsert(tk, "test_add_index", i, i+100)
-	}
+func TestAddIndexWorkerNum(t *testing.T) {
+	s, clean := createFailDBSuite(t)
+	defer clean()
 
-	is := s.dom.InfoSchema()
-	schemaName := model.NewCIStr("test_db")
-	tableName := model.NewCIStr("test_add_index")
-	tbl, err := is.TableByName(schemaName, tableName)
-	c.Assert(err, IsNil)
-
-	splitCount := 100
-	// Split table to multi region.
-	tableStart := tablecodec.GenTableRecordPrefix(tbl.Meta().ID)
-	s.cluster.SplitKeys(tableStart, tableStart.PrefixNext(), splitCount)
-
-	err = ddlutil.LoadDDLReorgVars(context.Background(), tk.Se)
-	c.Assert(err, IsNil)
-	originDDLAddIndexWorkerCnt := variable.GetDDLReorgWorkerCounter()
-	lastSetWorkerCnt := originDDLAddIndexWorkerCnt
-	atomic.StoreInt32(&ddl.TestCheckWorkerNumber, lastSetWorkerCnt)
-	ddl.TestCheckWorkerNumber = lastSetWorkerCnt
-	defer tk.MustExec(fmt.Sprintf("set @@global.tidb_ddl_reorg_worker_cnt=%d", originDDLAddIndexWorkerCnt))
-
-	if !s.IsCommonHandle { // only enable failpoint once
-		c.Assert(failpoint.Enable("github.com/pingcap/tidb/ddl/checkBackfillWorkerNum", `return(true)`), IsNil)
-		defer func() {
-			c.Assert(failpoint.Disable("github.com/pingcap/tidb/ddl/checkBackfillWorkerNum"), IsNil)
-		}()
+	tests := []struct {
+		name                   string
+		checkBackfillWorkerNum bool
```

Comment:
It seems this field is not used in the test, can we remove it?

Reply:
Nice catch. It is an intermediate variable. Removed.